frank/utilities.py

```diff
@@ -851,3 +853,72 @@ def convolve_profile(r, I, disc_i, disc_pa, clean_beam,
     I_smooth /= counts
 
     return I_smooth
+
+
+def generic_dht(x, f, Rmax=2.0, N=300, direction='forward', grid=None,
+                inc=0.0):
+    """
+    Compute the forward or backward Hankel transform of a function f(x)
+
+    Parameters
+    ----------
+    x : array, unit = [arcsec] or [lambda]
+        Radial or spatial frequency coordinates of f(x)
+    f : array, unit = [Jy / sr] or [Jy]
+        Amplitude values of f(x)
+    Rmax : float, unit = [arcsec], default=2.0
+        Maximum radius beyond which the real space function is zero
+    N : integer, default=1000
+        Number of terms to use in the Fourier-Bessel series
+    direction : { 'forward', 'backward' }, default='forward'
+        Direction of the transform. 'forward' is real space -> Fourier space.
+    grid : array, unit = [arcsec] or [lambda], default=None
+        The radial or spatial frequency points at which to sample the DHT.
+        If None, the DHT collocation points will be used.
+    inc : float, unit = [deg], default = 0.0
+        Source inclination. The total flux of the transform of f(x)
+        will be scaled by cos(inc); this has no effect if inc=0.
+
+    Returns
+    -------
+    grid : array, size=N, unit = [arcsec] or [lambda]
+        Spatial frequency or radial coordinates of the Hankel transform of f(x)
+    f_transform : array, size=N, unit = [Jy / sr] or [Jy]
+        Hankel transform of f(x)
+
+    """
+
+    if direction not in ['forward', 'backward']:
+        raise AttributeError("direction must be one of {}"
+                            "".format(['forward', 'backward']))
+
+    DHT = DiscreteHankelTransform(Rmax=Rmax / rad_to_arcsec, N=N, nu=0)
+    geom = FixedGeometry(inc, 0, 0, 0)
+    VM = VisibilityMapping(DHT, geom)
+
+    unit_convert = False
+
+    if direction == 'forward':
+        # map the profile f(x) onto the DHT collocation points.
+        # this is necessary for an accurate transform.
+        y = np.interp(DHT.r * rad_to_arcsec, x, f)
+
+        if grid is None:
+            grid = DHT.q
```

Comment:
```suggestion
            grid = VM.q
```

Reply:
change a 2nd DHT call to VM in generic_dht